### PR TITLE
batches: store encryption key "ID" alongside credentials

### PR DESCRIPTION
**Pull request by LawnGnome**
This is essentially what we do for `external_accounts`, and is required to handle the plaintext to encrypted upgrade scenario.

@eseliger, I think you're the only person who'll be affected by this in practice: you _may_ need to delete and recreate your local credentials. Sorry.

### CODE REVIEW

File: enterprise/internal/batches/store/integration_test.go [hunk 1]
```diff
@@ -16,21 +18,29 @@ func TestIntegration(t *testing.T) {
...
+		t.Run("BulkOperations", storeTest(db, nil, testStoreBulkOperations))
+
+		for name, key := range map[string]encryption.Key{
+			"no key":   nil,
+			"test key": &et.TestKey{},
+		} {
+			t.Run(name, func(t *testing.T) {
+				t.Run("SiteCredentials", storeTest(db, key, testStoreSiteCredentials))
+			})
+		}
```
Comment (eseliger) [inline enterprise/internal/batches/store/integration_test.go:44]: I wonder if we want to move this into that test fn so we don't clutter this function, but maybe not.

File: enterprise/internal/batches/store/site_credentials.go [hunk 2]
```diff
@@ -33,10 +33,17 @@ func (s *Store) CreateSiteCredential(ctx context.Context, c *btypes.SiteCredenti
 
 var createSiteCredentialQueryFmtstr = `
 -- source: enterprise/internal/batches/store/site_credentials.go:CreateSiteCredential
-INSERT INTO
-	batch_changes_site_credentials (external_service_type, external_service_id, credential, credential_enc, created_at, updated_at)
+INSERT INTO	batch_changes_site_credentials (
```
Comment (eseliger) [inline enterprise/internal/batches/store/site_credentials.go:36]:
```suggestion
INSERT INTO batch_changes_site_credentials (
```

**Review by eseliger — APPROVED** [review]
Thanks for the quick fix :)

File: enterprise/internal/batches/store/integration_test.go (hunk 1 above)
Comment (LawnGnome) [reply enterprise/internal/batches/store/integration_test.go:44]: I think we'll eventually want this for other types that get encrypted at rest, which is why I'm comfortable with it in the integration test. I don't feel super strongly about it, though.

File: enterprise/internal/batches/types/site_credential.go [hunk 3]
```diff
@@ -60,6 +65,14 @@ func (sc *SiteCredential) Authenticator(ctx context.Context) (auth.Authenticator
 // SetAuthenticator encrypts and sets the authenticator within the site
 // credential.
 func (sc *SiteCredential) SetAuthenticator(ctx context.Context, a auth.Authenticator) error {
+	// Set the key ID. This is cargo culted from external_accounts.go, and the
+	// key ID doesn't appear to be actually useful as anything other than a
```
Comment (mrnugget) [inline enterprise/internal/batches/types/site_credential.go:69]: I think in the future the key ID will be used to re-encrypt things with a new key, no?

**Comment by LawnGnome** [comment]
Closing; this needs a related, but different approach, and I don't want to confuse the code review history here.

**Pull request closed without merging**